projects/packages/my-jetpack/_inc/state/reducers.js

```diff
@@ -9,13 +9,29 @@ import { combineReducers } from '@wordpress/data';
 import {
 	SET_PURCHASES,
 	SET_PURCHASES_IS_FETCHING,
-	SET_PRODUCT_ACTION_ERROR,
+	SET_PRODUCT,
 	SET_PRODUCT_STATUS,
-	SET_FETCHING_PRODUCT_STATUS,
+	SET_IS_FETCHING_PRODUCT,
+	SET_PRODUCT_REQUEST_ERROR,
 } from './actions';
 
 const products = ( state = {}, action ) => {
 	switch ( action.type ) {
+		case SET_IS_FETCHING_PRODUCT: {
+			const { productId, isFetching } = action;
+			return {
+				...state,
+				isFetching: {
+					...state.isFetching,
+					[ productId ]: isFetching,
+				},
+				errors: {
+					...state.errors,
+					[ productId ]: isFetching ? undefined : state.errors[ productId ],
+				},
```

Comment:
This is starting to become really complex, mainly if we continue to create one key for each property/flag, would say that prefer to wrap everything inside product slug as key:

```javascript
products = {
  boost: {
    isFeching: false,
    errors: {},
    ...productData
  }
}
```

Reply:
I wouldn't like to pollute the item with these data that don't belong to the product by itself. The good thing about that is we can just pick the item data and pass them to the component, making the data propagation simpler and transparent, imo.

About complexity, I don't see significant pros and cons in between both approaches.

```
products: {
	backup: {
    	name,
		description,
		// more product data,
		isFetching: true,
		error: {
			code:...,
			message: ...,
			status: {}
		},
	},
	boost: {
		...
	},
	crm: {
		....
	}
}
```

```
products: {
	items: {
		backup: {
			name,
			description,
			// more product data,
		},
		boost: {
			...
		},
		crm: {
			...
		}
	}
	isFetching: {
		backup: true,
	},
	errors: {
		backup: {
			code:...,
			message: ...,
			status: {}
		},
	}
}
```

Reply:
You're right, looks the same :)

Reply:
Yes! Rephrasing, they look similar.